# HG changeset patch
# User Andrew John Hughes <gnu_andrew@member.fsf.org>
# Date 1339588177 -3600
# Node ID df0999b55509faf1504b3ff55152a61a7cff57fc
# Parent  e6d81fc2c9439652f903664d51a13732c73485ea
Make compile-for-7.patch apply after cacao/memory.patch

2012-06-13  Andrew John Hughes  <gnu_andrew@member.fsf.org>

	* Makefile.am:
	Apply compile-for-7-cacao-mem.patch instead of
	compile-for-7.patch when using a compiler that
	supports diamond and either USING_CACAO or BUILD_CACAO
	are set.
	* patches/boot/compile-for-7-cacao-mem.patch:
	Version of compile-for-7.patch post applying
	cacao/memory.patch.

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -364,8 +364,16 @@
 if JAVAC_LACKS_DIAMOND
 ICEDTEA_BOOT_PATCHES += patches/boot/ecj-diamond.patch
 else
+if USING_CACAO
+ICEDTEA_BOOT_PATCHES += patches/boot/compile-for-7-cacao-mem.patch
+else
+if BUILD_CACAO
+ICEDTEA_BOOT_PATCHES += patches/boot/compile-for-7-cacao-mem.patch
+else
 ICEDTEA_BOOT_PATCHES += patches/boot/compile-for-7.patch
 endif
+endif
+endif
 
 if DTDTYPE_QNAME
 ICEDTEA_BOOT_PATCHES += patches/boot/jaxws-getdtdtype.patch
diff --git a/patches/boot/compile-for-7-cacao-mem.patch b/patches/boot/compile-for-7-cacao-mem.patch
new file mode 100644
--- /dev/null
+++ b/patches/boot/compile-for-7-cacao-mem.patch
@@ -0,0 +1,42 @@
+diff -Nru openjdk-boot.orig/hotspot/make/linux/makefiles/rules.make openjdk-boot/hotspot/make/linux/makefiles/rules.make
+--- openjdk-boot.orig/hotspot/make/linux/makefiles/rules.make	2012-06-13 03:40:41.009311243 +0100
++++ openjdk-boot/hotspot/make/linux/makefiles/rules.make	2012-06-13 03:48:41.838257784 +0100
+@@ -139,8 +139,8 @@
+ RUN.JAR$(MAKE_VERBOSE) += >/dev/null
+ 
+ # Settings for javac
+-BOOT_SOURCE_LANGUAGE_VERSION = 6
+-BOOT_TARGET_CLASS_VERSION = 6
++BOOT_SOURCE_LANGUAGE_VERSION = 7
++BOOT_TARGET_CLASS_VERSION = 7
+ JAVAC_FLAGS = -g -encoding ascii -J-Xmx256m
+ BOOTSTRAP_JAVAC_FLAGS = $(JAVAC_FLAGS) -source $(BOOT_SOURCE_LANGUAGE_VERSION) -target $(BOOT_TARGET_CLASS_VERSION)
+ 
+diff -Nru openjdk-boot.orig/hotspot/make/solaris/makefiles/rules.make openjdk-boot/hotspot/make/solaris/makefiles/rules.make
+--- openjdk-boot.orig/hotspot/make/solaris/makefiles/rules.make	2012-06-13 03:40:41.009311243 +0100
++++ openjdk-boot/hotspot/make/solaris/makefiles/rules.make	2012-06-13 11:36:03.057522188 +0100
+@@ -131,8 +131,8 @@
+ RUN.JAR$(MAKE_VERBOSE) += >/dev/null
+ 
+ # Settings for javac
+-BOOT_SOURCE_LANGUAGE_VERSION = 6
+-BOOT_TARGET_CLASS_VERSION = 6
++BOOT_SOURCE_LANGUAGE_VERSION = 7
++BOOT_TARGET_CLASS_VERSION = 7
+ JAVAC_FLAGS = -g -encoding ascii -J-Xmx256m
+ BOOTSTRAP_JAVAC_FLAGS = $(JAVAC_FLAGS) -source $(BOOT_SOURCE_LANGUAGE_VERSION) -target $(BOOT_TARGET_CLASS_VERSION)
+ 
+diff -Nru openjdk-boot.orig/hotspot/make/windows/makefiles/rules.make openjdk-boot/hotspot/make/windows/makefiles/rules.make
+--- openjdk-boot.orig/hotspot/make/windows/makefiles/rules.make	2012-06-13 03:40:41.009311243 +0100
++++ openjdk-boot/hotspot/make/windows/makefiles/rules.make	2012-06-13 11:36:19.537797380 +0100
+@@ -43,8 +43,8 @@
+ !endif
+ 
+ # Settings for javac
+-BOOT_SOURCE_LANGUAGE_VERSION=6
+-BOOT_TARGET_CLASS_VERSION=6
++BOOT_SOURCE_LANGUAGE_VERSION=7
++BOOT_TARGET_CLASS_VERSION=7
+ JAVAC_FLAGS=-g -encoding ascii -J-Xmx256m
+ BOOTSTRAP_JAVAC_FLAGS=$(JAVAC_FLAGS) -source $(BOOT_SOURCE_LANGUAGE_VERSION) -target $(BOOT_TARGET_CLASS_VERSION)
+ 
